sycl/plugins/esimd_cpu/esimd_libcm.hpp

```diff
@@ -0,0 +1,327 @@
+                      WrappedLambda_ITEM_3DIM.get());
+
+    ArgVector.clear();
+  }
+
+  template <class ArgT = KernelArgType>
+  typename std::enable_if<std::is_same<ArgT, sycl::item<1, true>>::value>::type
+  run() {
+    auto WrappedLambda_ITEM_OFFSET_1DIM =
+        makeWrapper_ITEM_OFFSET_1DIM(ArgVector, MKernel);
+
+    ESimdCPUKernel ESimdCPU((fptrVoid)invokeLambda_ITEM_OFFSET_1DIM, SpaceDim);
+
+    ESimdCPU.launchMT(sizeof(struct LambdaWrapper_ITEM_OFFSET_1DIM),
+                      WrappedLambda_ITEM_OFFSET_1DIM.get());
+    ArgVector.clear();
+  }
+
+  template <class ArgT = KernelArgType>
+  typename std::enable_if<std::is_same<ArgT, sycl::item<2, true>>::value>::type
+  run() {
+    auto WrappedLambda_ITEM_OFFSET_2DIM =
+        makeWrapper_ITEM_OFFSET_2DIM(ArgVector, MKernel);
+
+    ESimdCPUKernel ESimdCPU((fptrVoid)invokeLambda_ITEM_OFFSET_2DIM, SpaceDim);
+
+    ESimdCPU.launchMT(sizeof(struct LambdaWrapper_ITEM_OFFSET_2DIM),
+                      WrappedLambda_ITEM_OFFSET_2DIM.get());
+    ArgVector.clear();
+  }
+
+  template <class ArgT = KernelArgType>
+  typename std::enable_if<std::is_same<ArgT, sycl::item<3, true>>::value>::type
+  run() {
+    auto WrappedLambda_ITEM_OFFSET_3DIM =
+        makeWrapper_ITEM_OFFSET_3DIM(ArgVector, MKernel);
+
+    ESimdCPUKernel ESimdCPU((fptrVoid)invokeLambda_ITEM_OFFSET_3DIM, SpaceDim);
+
+    ESimdCPU.launchMT(sizeof(struct LambdaWrapper_ITEM_OFFSET_3DIM),
+                      WrappedLambda_ITEM_OFFSET_3DIM.get());
+    ArgVector.clear();
+  }
+
+  template <class ArgT = KernelArgType>
+  typename std::enable_if<std::is_same<ArgT, sycl::nd_item<1>>::value>::type
+  run() {
+    auto WrappedLambda_NDITEM_1DIM =
+        makeWrapper_NDITEM_1DIM(ArgVector, MKernel);
+
+    ESimdCPUKernel ESimdCPU((fptrVoid)invokeLambda_NDITEM_1DIM, SpaceDim);
+
+    ESimdCPU.launchMT(sizeof(struct LambdaWrapper_NDITEM_1DIM),
+                      WrappedLambda_NDITEM_1DIM.get());
+    ArgVector.clear();
+  }
+
+  template <class ArgT = KernelArgType>
+  typename std::enable_if<std::is_same<ArgT, sycl::nd_item<2>>::value>::type
+  run() {
+    auto WrappedLambda_NDITEM_2DIM =
+        makeWrapper_NDITEM_2DIM(ArgVector, MKernel);
+
+    ESimdCPUKernel ESimdCPU((fptrVoid)invokeLambda_NDITEM_2DIM, SpaceDim);
+
+    ESimdCPU.launchMT(sizeof(struct LambdaWrapper_NDITEM_2DIM),
+                      WrappedLambda_NDITEM_2DIM.get());
+    ArgVector.clear();
+  }
+
+  template <class ArgT = KernelArgType>
+  typename std::enable_if<std::is_same<ArgT, sycl::nd_item<3>>::value>::type
+  run() {
+    auto WrappedLambda_NDITEM_3DIM =
+        makeWrapper_NDITEM_3DIM(ArgVector, MKernel);
+
+    ESimdCPUKernel ESimdCPU((fptrVoid)invokeLambda_NDITEM_3DIM, SpaceDim);
+
+    ESimdCPU.launchMT(sizeof(struct LambdaWrapper_NDITEM_3DIM),
+                      WrappedLambda_NDITEM_3DIM.get());
+    ArgVector.clear();
+  }
+};
+
+inline void __cm_mt_barrier() { cm_support::mt_barrier(); }
```

Comment:
Those APIs don't seem to belong here - the kernel invocation core. Can they be moved to some other source like esimd_intrinsics_impl.h?

Reply:
Some "__cm" functions are called from ESIMD intrincis. If it is okay to call "cm_support::" functions directly from those intrinsics, these wrapping functions can be removed.